R/group_apply.R

```diff
@@ -0,0 +1,133 @@
+#' Apply a function over grouping(s) of data
+#'
+#' A function to apply `palaeoverse` functionality across subsets (groups) of
+#' data, delineated using one or more variables. Functions which receive a
+#' `data.frame` as input (e.g. `nrow`, `ncol`, `lengths`, `unique`) may also be
+#' used.
+#'
+#' @param occdf \code{dataframe}. A dataframe of fossil occurrences or taxa,
+#' as relevant to the desired function.
+#' This dataframe must contain the grouping variables and the necessary
+#' variables for the function you wish to call (see function-specific
+#' documentation for required columns).
+#' @param group \code{character}. A vector of column names, specifying the
+#' desired subgroups (e.g. "collection_no", "stage_bin"). Supplying more than
+#' one grouping variable will produce an output containing subgroups for each
+#' unique combination of values.
+#' @param fun \code{function}. The function you wish to apply to
+#' `occdf`. See details for compatible functions.
+#' @param ... Additional arguments available in the called
+#' function. These additional arguments may be required for function arguments
+#' without default values, or if you wish to overwrite the default argument
+#' value (see examples).
+#'
+#' @return A \code{dataframe} of the outputs from the selected function, with
+#' appended column(s) indicating the user-defined groups (`grouping`).
```

Comment:
```suggestion
#' appended column(s) indicating the user-defined groups.
```
This can be removed now.